doc/main/reference/task_group_extensions.rst

```diff
@@ -62,43 +60,19 @@ Synopsis
 Member Functions
 ----------------
 
-.. cpp:function:: task_group(task_group_context& context)
-
-Constructs an empty ``task_group``, which tasks are associated with the ``context``.
-
-
 .. cpp:function:: template<typename F> task_handle  defer(F&& f)
 
-Creates a deferred task to compute ``f()`` and returns ``task_handle`` pointing to it.
-
-The task is not scheduled for execution until explicitly requested. For example, with the ``task_group::run`` method.
-However, the task is still added into the ``task_group``, thus the ``task_group::wait`` method waits until the ``task_handle`` is either scheduled or destroyed.
-
-The ``F`` type must meet the `Function Objects` requirements described in the [function.objects] section of the ISO C++ Standard.
-
 As an optimization hint, ``F`` might return a ``task_handle``, which task object can be executed next.
 
 .. note::
    The ``task_handle`` returned by the function must be created with ``*this`` ``task_group``. It means, with the one for which run method is called, otherwise it is an undefined behavior. 
 
-**Returns:** ``task_handle`` object pointing to task to compute ``f()``.
-
-
-.. cpp:function:: void run(task_handle&& h)
-
-Schedules the task object pointed by the ``h`` for execution.
-
-.. caution:: If ``h`` is empty or ``*this`` is not the same ``task_group`` that ``h`` is created with, the behavior is undefined.
-
+.. cpp:function:: template<typename F> task_group_status run_and_wait(const F& f)
 
-.. cpp:function:: task_group_status run_and_wait(task_handle&& h)
-
-Equivalent to ``{run(std::move(h)); return wait();}``.
- 
-**Returns**: The status of ``task_group``.
+As an optimization hint, ``F`` might return a ``task_handle``, which task object can be executed next.
 
-.. caution::
-   If ``h`` is empty or ``*this`` is not the same ``task_group`` that ``h`` is created with, the behavior is undefined.
+.. note::
+   The ``task_handle`` returned by the function must be created with ``*this`` ``task_group``. It means, with the one for which run method is called, otherwise it is an undefined behavior. 
```

Comment:
```suggestion
   The ``task_handle`` returned by the function must be created using ``*this`` ``task_group``. It means, with the one for which the run method is called, otherwise it is undefined behavior. 
```